LLK/scratch-vm
# Prevent console errors when disconnecting/reconnect to Vernier extension

### Resolves

https://github.com/LLK/scratch-vm/issues/1908

### Proposed Changes

This PR does two things: (1) it keeps track of when the measurements have actually been enabled (before, it only checked when Scratch Link had first connect to the device) and (2) it prevents sending 'close' to a device which has already been closed (by clearing the internal `this._device` after closing.

### Reason for Changes

The extension works find in the gui without these changes, however, it produces errors in the console.

### Test Coverage
- Add the GDX-FOR extension, and dismiss the connection modal
- Check the monitor checkbox for force (or add hat block to the workspace, or run a forever loop containing a sensor reporter)
- Connect to the device, and the error appears during the connection process
